firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -169,63 +174,77 @@ private void removeMutationBatch(String uid, int batchId) {
-    // A cache of targets and associated metadata
-    db.execSQL(
-        "CREATE TABLE targets ("
-            + "target_id INTEGER PRIMARY KEY, "
-            + "canonical_id TEXT, "
-            + "snapshot_version_seconds INTEGER, "
-            + "snapshot_version_nanos INTEGER, "
-            + "resume_token BLOB, "
-            + "last_listen_sequence_number INTEGER,"
-            + "target_proto BLOB)");
-
-    db.execSQL("CREATE INDEX query_targets ON targets (canonical_id, target_id)");
-
-    // Global state tracked across all queries, tracked separately
-    db.execSQL(
-        "CREATE TABLE target_globals ("
-            + "highest_target_id INTEGER, "
-            + "highest_listen_sequence_number INTEGER, "
-            + "last_remote_snapshot_version_seconds INTEGER, "
-            + "last_remote_snapshot_version_nanos INTEGER)");
+    if (!tableExists("targets")) {
+      // A cache of targets and associated metadata
+      db.execSQL(
+          "CREATE TABLE targets ("
+              + "target_id INTEGER PRIMARY KEY, "
+              + "canonical_id TEXT, "
+              + "snapshot_version_seconds INTEGER, "
+              + "snapshot_version_nanos INTEGER, "
+              + "resume_token BLOB, "
+              + "last_listen_sequence_number INTEGER,"
+              + "target_proto BLOB)");
+
+      db.execSQL("CREATE INDEX query_targets ON targets (canonical_id, target_id)");
+
+      // Global state tracked across all queries, tracked separately
+      db.execSQL(
+          "CREATE TABLE target_globals ("
+              + "highest_target_id INTEGER, "
+              + "highest_listen_sequence_number INTEGER, "
+              + "last_remote_snapshot_version_seconds INTEGER, "
+              + "last_remote_snapshot_version_nanos INTEGER)");
 
-    // A Mapping table between targets and document paths
-    db.execSQL(
-        "CREATE TABLE target_documents ("
-            + "target_id INTEGER, "
-            + "path TEXT, "
-            + "PRIMARY KEY (target_id, path))");
+      // A Mapping table between targets and document paths
+      db.execSQL(
+          "CREATE TABLE target_documents ("
+              + "target_id INTEGER, "
+              + "path TEXT, "
+              + "PRIMARY KEY (target_id, path))");
 
-    // The document_targets reverse mapping table is just an index on target_documents.
-    db.execSQL("CREATE INDEX document_targets ON target_documents (path, target_id)");
+      // The document_targets reverse mapping table is just an index on target_documents.
+      db.execSQL("CREATE INDEX document_targets ON target_documents (path, target_id)");
+    }
   }
 
   private void dropQueryCache() {
-    db.execSQL("DROP TABLE targets");
-    db.execSQL("DROP TABLE target_globals");
-    db.execSQL("DROP TABLE target_documents");
+    // This might be overkill, but if any future migration drops these, it's possible we could try
+    // dropping tables that don't exist.
```

Comment:
If we had to put a version on it, it would probably be `V1`, since that's the only query cache we create. I think I'm in favor of that, since if we have a `V2` query cache, it could be completely separate tables, and the distinction would be useful.